pkg/vm/vm.go

```diff
@@ -93,6 +96,11 @@ func (vm *VirtualMachine) RunStep(hintRunner HintRunner) error {
 		return fmt.Errorf("cannot decode step at %d: %w", vm.Context.Pc, err)
 	}
 
+	// store the trace before state change
+	if vm.config.ProofMode {
+		vm.Trace = append(vm.Trace, vm.Context)
```

Comment:
What's the cost of appending here? Is resizing cheap or should we make the slice big enough at the beginning?

Reply:
Resizing cost scales linearly with the size of the array. For each resize you have to:
* Allocate new memory
* Copy the old memory to the new valory

Different compilers use different heuristics for resizing , the the cost becomes an issue when you have to resize bigger arrays. I think the go rules are the following:
* if smaller than 1024, duplicate it's size
* otherwise, increase its size by 25 percent

In this case is hard to predict if a language will be a short or a long run (the halting problem ofc) and I think that initial cost of resizing is negligible enough at the beginning. That does not stopping from giving it an initial value, but I don't think it would help us much

Reply:
Understood! Let's leave it as is